bee-node/src/config.rs

```diff
@@ -232,18 +231,31 @@ mod test {
 
     #[test]
     fn config_files_conformity() -> Result<(), NodeConfigError> {
+        let _ = NodeConfigBuilder::<Storage>::from_file(concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/config.chrysalis-comnet.json"
+        ))?;
         let _ = NodeConfigBuilder::<Storage>::from_file(concat!(
             env!("CARGO_MANIFEST_DIR"),
             "/config.chrysalis-comnet.toml"
         ))?;
+        let _ = NodeConfigBuilder::<Storage>::from_file(concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/config.chrysalis-devnet.json"
+        ))?;
         let _ = NodeConfigBuilder::<Storage>::from_file(concat!(
             env!("CARGO_MANIFEST_DIR"),
             "/config.chrysalis-devnet.toml"
         ))?;
+        let _ = NodeConfigBuilder::<Storage>::from_file(concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/config.chrysalis-mainnet.json"
+        ))?;
         let _ = NodeConfigBuilder::<Storage>::from_file(concat!(
             env!("CARGO_MANIFEST_DIR"),
             "/config.chrysalis-mainnet.toml"
         ))?;
```

Comment:
Could we `assert_eq!` here? For example `config.chrysalis-comnet.json` and `config.chrysalis-comnet.toml` should produce the same `NodeConfigBuilder`.

Reply:
Good idea but it will force us to derive Eq/PartialEq on *all* config objects

Reply:
I can feature gate it under tests thou

Reply:
I actually can't do that atm because some of the nested types don't implement PartialEq and they are not ours